### PR TITLE
reduce clang-14 warnings

### PR DESCRIPTION
**Pull request by ZuseZ4**

### CODE REVIEW

**Comment by ZuseZ4** [comment]
@tgymnich only actual change is in [InstructionBatcher.cpp](https://github.com/EnzymeAD/Enzyme/pull/1040/files#diff-8f474a5d6163109845a750e234c464997a806c2d502f13c395e59aeed3a2ae1a), before it was UB I guess (but maybe worked by coincidence), now it would panic.

File: enzyme/Enzyme/GradientUtils.cpp [hunk 1]
```diff
@@ -4124,6 +4124,9 @@ Constant *GradientUtils::GetOrCreateShadowFunction(
     return ConstantExpr::getPointerCast(GV, fn->getType());
   }
   }
+  llvm::errs() << "unhandled mode\n";
```
Comment (tgymnich) [inline enzyme/Enzyme/GradientUtils.cpp:4127]: why is this necessary? We should already handle all the cases.

Comment (wsmoses) [reply enzyme/Enzyme/GradientUtils.cpp:4127]: @ZuseZ4 remove the llvm errs warning and change the llvm_unreachable text to "Illegal state: unknown mode for GetOrCreateShadowFunction

@tgymnich my guess is that some compilers cannot prove that we handle all cases of the enum, and this is seen as a path without a return/unreachable.

**Review by wsmoses — APPROVED** [review]

File: enzyme/Enzyme/GradientUtils.cpp (hunk 1 above)
Comment (ZuseZ4) [reply enzyme/Enzyme/GradientUtils.cpp:4127]: Indeed, for clang 14.0.6 it's not enough to handle all valid enum states in the places I updated.
At least this fixed the warning for it.

I'll adjust the message and correct tablegen tmrw.
